solidity/contracts/deposit/Deposit.sol

```diff
@@ -385,89 +411,230 @@ contract Deposit is DepositFactoryAuthority {
+        bytes8 _newOutputValueBytes
+    ) public {
+        self.increaseRedemptionFee(_previousOutputValueBytes, _newOutputValueBytes);
     }
 
-    /// @notice     Goes from courtesy call to active.
-    /// @dev        Only callable if collateral is sufficient and the deposit is not expiring.
-    function exitCourtesyCall() public {
-        self.exitCourtesyCall();
+    /// @notice Anyone may submit a redemption proof to the deposit showing that
+    ///         a transaction was submitted and sufficiently confirmed on the
+    ///         Bitcoin chain transferring the deposit lot size's amount of BTC
+    ///         from the signer-controlled private key corresponding to this
+    ///         deposit to the requested redemption output script. This will
+    ///         move the deposit into a redeemed state.
+    /// @dev Takes a pre-parsed transaction and calculates values needed to
+    ///      verify funding. Signers can have their bonds seized if this is not
+    ///      called within `TBTCConstants.REDEMPTION_PROOF_TIMEOUT` seconds of
+    ///      a redemption signature being provided.
+    /// @param _txVersion Transaction version number (4-byte little-endian).
+    /// @param _txInputVector All transaction inputs prepended by the number of
+    ///        inputs encoded as a VarInt, max 0xFC(252) inputs.
+    /// @param _txOutputVector All transaction outputs prepended by the number
+    ///         of outputs encoded as a VarInt, max 0xFC(252) outputs.
+    /// @param _txLocktime Final 4 bytes of the transaction.
+    /// @param _merkleProof The merkle proof of transaction inclusion in a
+    ///        block.
+    /// @param _txIndexInBlock Transaction index in the block (0-indexed).
+    /// @param _bitcoinHeaders Single bytestring of 80-byte bitcoin headers,
+    ///        lowest height first.
+    function provideRedemptionProof(
+        bytes4 _txVersion,
+        bytes memory _txInputVector,
+        bytes memory _txOutputVector,
+        bytes4 _txLocktime,
+        bytes memory _merkleProof,
+        uint256 _txIndexInBlock,
+        bytes memory _bitcoinHeaders
+    ) public {
+        self.provideRedemptionProof(
+            _txVersion,
+            _txInputVector,
+            _txOutputVector,
+            _txLocktime,
+            _merkleProof,
+            _txIndexInBlock,
+            _bitcoinHeaders
+        );
     }
 
-    /// @notice     Notify the contract that the signers are undercollateralized.
-    /// @dev        Calls out to the system for oracle info.
-    function notifyUndercollateralizedLiquidation() public {
-        self.notifyUndercollateralizedLiquidation();
+//--------------------------- MUTATING HELPERS -------------------------------//
+
+    /// @notice This function can only be called by the deposit factory; use
+    ///         `DepositFactory.createDeposit` to create a new deposit.
+    /// @dev Initializes a new deposit clone with the base state for the
```

Comment:
Should `@notice` explain the thing and `@dev` have developer details?

Reply:
Same as above.